rotkehlchen/chain/evm/decoding/beefy_finance/utils.py

```diff
@@ -91,36 +91,46 @@ def _process_beefy_vault(
         description='Querying Beefy finance vaults',
         should_notify=False,
     )
+    name = symbol = decimals = None
     if (vault_type := vault.get('type')) is None or vault_type in ('standard', 'gov'):
         # in boost (no type key), standard, and gov vaults the `tokenAddress` is the vault address,
         # so add it as the earned token's underlying token to be used when querying the price.
         if (raw_token_address := vault.get('tokenAddress')) is not None:
-            token_address = get_or_create_evm_token(
+            token = get_or_create_evm_token(
                 userdb=database,
                 evm_inquirer=evm_inquirer,
                 evm_address=deserialize_evm_address(raw_token_address),
                 chain_id=evm_inquirer.chain_id,
                 encounter=encounter,
-            ).evm_address
+            )
         else:
             # For vaults without tokenAddress field (native token vaults like ETH, BNB, MATIC),
             # use the chain's wrapped native token as the underlying asset
-            token_address = evm_inquirer.wrapped_native_token.evm_address
+            token = evm_inquirer.wrapped_native_token
 
         underlying_tokens = [UnderlyingToken(
-            address=token_address,
+            address=token.evm_address,
             token_kind=TokenKind.ERC20,
             weight=ONE,
         )]
+        decimals = token.decimals
     elif vault_type == 'cowcentrated':
         underlying_tokens = None  # CLM vault where the earned token is the vault token itself.
     else:
         log.warning(f'Unknown Beefy vault type {vault_type} for vault {vault}')
         return
 
+    if vault.get('version') == 1:  # legacy beefy boost vaults are not standard ERC20 tokens but implement balanceOf, requiring manual token metadata creation  # noqa: E501
+        capitalized_pool_parts = [i.capitalize() for i in vault['poolId'].split('-')]
+        name = f'Reward Moo {" ".join(capitalized_pool_parts)}'
+        symbol = f'rmoo{"".join(capitalized_pool_parts)}'
```

Comment:
what if they change the format for the names? is this how they show them in the ui? worst case it will split the names differently i guess

Reply:
> what if they change the format for the names? is this how they show them in the ui? worst case it will split the names differently i guess

that's how it's shown in the UI and how we also query their prices